src/tests/ftest/io/eight_servers_mpiio.py

```diff
@@ -24,77 +24,37 @@
 from __future__ import print_function
 
 import os
-import sys
-import json
-from avocado import Test
-
-sys.path.append('./util')
-sys.path.append('../util')
-sys.path.append('../../../utils/py')
-sys.path.append('./../../utils/py')
-import server_utils
 import write_host_file
 import ior_utils
+
+from apricot import TestWithServers
 from mpio_utils import MpioUtils, MpioFailed
-from daos_api import DaosContext, DaosPool, DaosApiError
+from daos_api import DaosPool, DaosApiError
 
-class EightServers(Test):
+class EightServers(TestWithServers):
     """
     Test class Description: Runs IOR with 8 servers.
-
+    :avocado: recursive
     """
 
-    def __init__(self, *args, **kwargs):
-
-        super(EightServers, self).__init__(*args, **kwargs)
+    def setUp(self):
+        super(EightServers, self).setUp()
 
-        self.basepath = None
-        self.server_group = None
-        self.context = None
-        self.pool = None
-        self.num_procs = None
-        self.hostlist_servers = None
-        self.hostfile_servers = None
-        self.hostlist_clients = None
-        self.hostfile_clients = None
+        # set required variables
         self.mpio = None
-
-    def setUp(self):
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-        print("<<{}>>".format(self.basepath))
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-
-        self.hostlist_servers = self.params.get("test_servers",
-                                                '/run/hosts/test_machines/*')
-        self.hostfile_servers = (
-            write_host_file.write_host_file(self.hostlist_servers,
-                                            self.workdir))
-        print("Host file servers is: {}".format(self.hostfile_servers))
-
-        self.hostlist_clients = self.params.get("test_clients",
-                                                '/run/hosts/test_machines/*')
         self.num_procs = self.params.get("np", '/run/ior/client_processes/*')
         self.hostfile_clients = (
             write_host_file.write_host_file(self.hostlist_clients, self.workdir,
                                             None))
-        print("Host file clients is: {}".format(self.hostfile_clients))
-
-        server_utils.run_server(self.hostfile_servers, self.server_group,
-                                self.basepath)
 
     def tearDown(self):
         try:
-            if self.pool is not None and self.pool.attached:
+            if self.hostfile_clients is not None:
+                os.remove(self.hostfile_clients)
```

Comment:
You don't need to remove this file as it's created in the avocado _workdir_.

Reply:
Done